runtime/drivers/athena/sql_store.go

```diff
@@ -0,0 +1,270 @@
+		cleanupFn:    cleanupFn,
+	}, nil
+}
+
+func (c *Connection) awsConfig(ctx context.Context, awsRegion string) (aws.Config, error) {
+	loadOptions := []func(*config.LoadOptions) error{
+		// Setting the default region to an empty string, will result in the default region value being ignored
+		config.WithDefaultRegion("us-east-1"),
+		// Setting the region to an empty string, will result in the region value being ignored
+		config.WithRegion(awsRegion),
+	}
+
+	// If one of the static properties is specified: access key, secret key, or session token, use static credentials,
+	// Else fallback to the SDK's default credential chain (environment, instance, etc) unless AllowHostAccess is false
+	if c.config.AccessKeyID != "" || c.config.SecretAccessKey != "" || c.config.SessionToken != "" {
+		p := credentials.NewStaticCredentialsProvider(c.config.AccessKeyID, c.config.SecretAccessKey, c.config.SessionToken)
+		loadOptions = append(loadOptions, config.WithCredentialsProvider(p))
+	} else if !c.config.AllowHostAccess {
+		return aws.Config{}, fmt.Errorf("static creds are not provided, and host access is not allowed")
+	}
+
+	return config.LoadDefaultConfig(ctx, loadOptions...)
+}
+
+func (c *Connection) unload(ctx context.Context, client *athena.Client, conf *sourceProperties, unloadLocation string) error {
+	finalSQL := fmt.Sprintf("UNLOAD (%s) TO '%s' WITH (format = 'PARQUET')", conf.SQL, unloadLocation)
+
+	executeParams := &athena.StartQueryExecutionInput{
+		QueryString: aws.String(finalSQL),
+	}
+
+	// If output_location is set, use it and don't set workgroup because the workgroup can override the output location
+	// Otherwise use specified workgroup or the "primary" workgroup
+	// see https://docs.aws.amazon.com/athena/latest/ug/querying.html
+	if conf.OutputLocation != "" {
+		executeParams.ResultConfiguration = &types2.ResultConfiguration{
+			OutputLocation: aws.String(conf.OutputLocation),
+		}
+	} else {
+		workgroup := conf.Workgroup
+		if workgroup == "" {
+			// fallback to "primary" (default) workgroup if no workgroup is specified
+			workgroup = "primary"
+		}
+		executeParams.WorkGroup = aws.String(workgroup)
+	}
+
+	queryExecutionOutput, err := client.StartQueryExecution(ctx, executeParams)
+	if err != nil {
+		return err
+	}
+
+	tm := time.NewTimer(5 * time.Minute)
+	defer tm.Stop()
+	for {
+		select {
+		case <-tm.C:
+			return fmt.Errorf("Athena ingestion timed out")
+		default:
+			status, err := client.GetQueryExecution(ctx, &athena.GetQueryExecutionInput{
+				QueryExecutionId: queryExecutionOutput.QueryExecutionId,
+			})
+			if err != nil {
+				return err
+			}
+
+			switch status.QueryExecution.Status.State {
+			case types2.QueryExecutionStateSucceeded, types2.QueryExecutionStateCancelled:
+				return nil
+			case types2.QueryExecutionStateFailed:
+				return fmt.Errorf("Athena query execution failed %s", *status.QueryExecution.Status.AthenaError.ErrorMessage)
+			}
+		}
+		time.Sleep(time.Second)
+	}
```

Comment:
1. Instead of using a hard-coded timer, it can use the `ctx` (check `<-ctx.Done()`), which has a timeout that's configurable through the `timeout:` YAML property.
2. If there's a `ctx` cancellation/timeout, it would be nice to cancel the running query.
3. For `case ..., types2.QueryExecutionStateCancelled` – shouldn't this case return an error? Otherwise, it will try to consume the results of the cancelled query.